pandas/core/categorical.py

```diff
@@ -881,6 +881,25 @@ def notnull(self):
         """
         return ~self.isnull()
 
+    def dropna(self, inplace=False):
+        """
+        Return the Categorical without null values.
+
+        Both missing values (-1 in .codes) and NA as a category are detected.
```

Comment:
by definition this should remove nan from the categories (that's the point of the method). I don't like inplace in general and not necessary here, pls take it out. ``_update_inplace`` doesn't exist its a generic method. (so clearly this is not tested).